src/main/java/edu/harvard/iq/dataverse/api/auth/BearerTokenAuthMechanism.java

```diff
@@ -0,0 +1,117 @@
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken;
+import edu.harvard.iq.dataverse.UserServiceBean;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.authorization.UserRecordIdentifier;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.OAuth2Exception;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc.OIDCAuthProvider;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.authorization.users.User;
+import edu.harvard.iq.dataverse.settings.FeatureFlags;
+
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.HttpHeaders;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+public class BearerTokenAuthMechanism implements AuthMechanism {
+    private static final String BEARER_AUTH_SCHEME = "Bearer";
+    public static final String UNAUTHORIZED_BEARER_TOKEN = "Unauthorized bearer token";
+    public static final String INVALID_BEARER_TOKEN = "Could not parse bearer token";
+    public static final String BEARER_TOKEN_DETECTED_NO_OIDC_PROVIDER_CONFIGURED = "Bearer token detected, no OIDC provider configured";
+
+    @Inject
+    protected AuthenticationServiceBean authSvc;
+    @Inject
+    protected UserServiceBean userSvc;
+    private static final Logger logger = Logger.getLogger(BearerTokenAuthMechanism.class.getCanonicalName());
+    @Override
+    public User findUserFromRequest(ContainerRequestContext containerRequestContext) throws WrappedAuthErrorResponse {
+        if (FeatureFlags.API_BEARER_AUTH.enabled()) {
+            Optional<String> bearerToken = getRequestApiKey(containerRequestContext);
+            // No Bearer Token present, hence no user can be authenticated
+            if (!bearerToken.isPresent()) {
+                return null;
+            }
+            // Validate and verify provided Bearer Token, and retrieve UserRecordIdentifier
+            // TODO: Get the identifier from an invalidating cache to avoid lookup bursts of the same token. Tokens in the cache should be removed after some (configurable) time.
+            UserRecordIdentifier userInfo = verifyOidcBearerTokenAndGetUserIndentifier(bearerToken.get());
+
+            // retrieve Authenticated User from AuthService
+            AuthenticatedUser authUser = authSvc.lookupUser(userInfo);
+            if (authUser != null) {
+                // track the API usage
+                authUser = userSvc.updateLastApiUseTime(authUser);
+                return authUser;
+            } else {
+                // a valid Token was presented, but we have no associated user account.
+                logger.log(Level.WARNING, "Bearer  token detected, OIDC provider {0} validated Token but no linked UserAccount", userInfo.getUserRepoId());
```

Comment:
```suggestion
                logger.log(Level.WARNING, "Bearer token detected, OIDC provider {0} validated Token but no linked UserAccount", userInfo.getUserRepoId());
                // TODO: Instead of returning null, we should throw a meaningful error to the client.
                // Probably this will be a wrapped auth error response with an error code and a string describing the problem.
```

@poikilotherm says you're willing to help with this. (We wrote this suggestion together.)